### PR TITLE
Bugfix/restore handle quote error headers

### PR DESCRIPTION
**Pull request by vbarzokas**
modify/set the headers in `sendErrorCallback` only in case it is explicitly requested to do so, as this function is called in two cases:
      // 1. (do not modify them) when the Switch needs to relay an error, e.g. from a DFSP to another
      // 2. (modify/set them) when the Switch needs to send errors that are originating in the Switch, e.g. to send an error back to the caller

Note: I have run the `npm audit fix` command in order to automatically fix any vulnerabilities, as there were thousands of them in the existing state. After that there are still 6 _moderate_, which need manual fix and cannot be fixed automatically.

### CODE REVIEW

File: src/model/quotes.js [hunk 1]
```diff
@@ -845,7 +845,7 @@ class QuotesModel {
    *
    * @returns {promise}
    */
-  async sendErrorCallback (fspiopSource, fspiopError, quoteId, headers, span) {
+  async sendErrorCallback (fspiopSource, fspiopError, quoteId, headers, span, modifyHeaders = true) {
```
Comment (rmothilal) [inline src/model/quotes.js]: should this not be false?

File: src/model/quotes.js [hunk 2]
```diff
@@ -611,7 +611,7 @@ class QuotesModel {
         // we didnt get an endpoint for the payee dfsp!
         // make an error callback to the initiator
         const fspiopError = ErrorHandler.CreateFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.DESTINATION_FSP_ERROR, `No FSPIOP_CALLBACK_URL_QUOTES found for quote ${quoteId} PAYER party`, null, fspiopSource)
-        return this.sendErrorCallback(fspiopSource, fspiopError, quoteId, headers)
+        return this.sendErrorCallback(fspiopSource, fspiopError, quoteId, headers, false)
```
Comment (bushjames) [inline src/model/quotes.js]: @vbarzokas I think this is a case where we are sending error back to caller...so modifyHeaders should be "true" here.

File: src/model/quotes.js [hunk 3]
```diff
@@ -709,7 +709,7 @@ class QuotesModel {
       const fspiopError = ErrorHandler.CreateFSPIOPErrorFromErrorInformation(error)
 
       // Needed to add await here to prevent 'span already finished' bug
-      await this.sendErrorCallback(headers[ENUM.Http.Headers.FSPIOP.DESTINATION], fspiopError, quoteId, headers, span)
+      await this.sendErrorCallback(headers[ENUM.Http.Headers.FSPIOP.DESTINATION], fspiopError, quoteId, headers, span, true)
```
Comment (bushjames) [inline src/model/quotes.js]: @vbarzokas this is the one where we DONT want to modify headers.

**Comment by rmothilal** [comment]
please use     "audit:resolve": "SHELL=sh resolve-audit",
script in the package.json to fix or ignore the issue

**Pull request merged**